src/array.rs

```diff
@@ -1036,6 +1036,112 @@ impl<T: Element, D: Dimension> PyArray<T, D> {
     }
 }
 
+#[cfg(feature = "nalgebra")]
+impl<N, D> PyArray<N, D>
+where
+    N: nalgebra::Scalar + Element,
+    D: Dimension,
+{
+    fn try_as_matrix_shape_strides<R, C, RStride, CStride>(
+        &self,
+    ) -> Option<((R, C), (RStride, CStride))>
+    where
+        R: nalgebra::Dim,
+        C: nalgebra::Dim,
+        RStride: nalgebra::Dim,
+        CStride: nalgebra::Dim,
+    {
+        let ndim = self.ndim();
+        let shape = self.shape();
+        let strides = self.strides();
+
+        if ndim != 1 && ndim != 2 {
+            return None;
+        }
+
+        if strides.iter().any(|strides| *strides < 0) {
+            return None;
+        }
+
+        let rows = shape[0];
+        let cols = *shape.get(1).unwrap_or(&1);
+
+        if R::try_to_usize().map(|expected| rows == expected) == Some(false) {
```

Comment:
I don't quite know what `R::try_to_usize() == None` mean, but is it OK to ignore it?

Reply:
This case happens for the `Dynamic` dimensions which can store any `usize` and so one cannot tell their value just based on the type. (Whereas for `Const<N>` one always has `Const::<N>::try_to_usize() == Some(N)`.)

In this case, it means that the return uses a `Dynamic` dimension for rows/columns/row strides/column strides and will handle any value so that the check can be ignored. Only if the given type has a fixed value it must either match (`Some(true)`) or the conversion fails (`Some(false)`).